core/cap-0046-02.md

```diff
@@ -60,61 +57,60 @@ This CAP defines two possible kinds of contract sources:
+expiration CAP).
 
 A contract instance may own an arbitrary amount of ledger entries attributed to
 its identifier. Contracts that share the same source in no way may influence
 each other; from the perspective of a contract invoker there is no difference
 between calling the contracts with the same or different source references
 (besides the possible contract-defined behavior differences).
 
-### Contract identifier preimage types
-
-This CAP defines the following supported contract identifier preimage kinds. The
-use cases for every identifier type are described in the following sections.
-Every preimage is a part of the `HashIDPreimage` union and has a unique tag
-associated with it in order to ensure that there are no collisions with other
-hashes in the protocol.
-
-- `ENVELOPE_TYPE_CONTRACT_ID_FROM_ED25519`: built from an `ed25519` public key
-  and the user-specified `uint256` salt.
-- `ENVELOPE_TYPE_CONTRACT_ID_FROM_SOURCE_ACCOUNT`: built from a Stellar account
-  identifier and the user-specified `uint256` salt.
-- `ENVELOPE_TYPE_CONTRACT_ID_FROM_ASSET`: built from a Stellar `Asset`
+### Contract identifier preimage type
+
+`ENVELOPE_TYPE_CONTRACT_ID` is the `HashIDPreimage` type used for contract
+identifiers. This unique tag is what ensures that there are no collisions with
+other hashes in the protocol. It contains the `Hash` of the `networkID` to
+ensure that every network has unique set of contract identifiers, along with a
+`ContractIDPreimage`, which is a union that supports both
+`CONTRACT_ID_PREIMAGE_FROM_ADDRESS` and `CONTRACT_ID_PREIMAGE_FROM_ASSET`.
+- `CONTRACT_ID_PREIMAGE_FROM_ADDRESS`: built from an an `SCAddress` and the
+  user-specified `uint256` salt.
+- `CONTRACT_ID_PREIMAGE_FROM_ASSET`: built from a Stellar `Asset`
   structure.
-- `ENVELOPE_TYPE_CONTRACT_ID_FROM_CONTRACT`: built from another contract
-  identifier and contract-specified `uint256` salt.
-
-Every preimage must also include a `networkID` (a hash of the network
-passphrase) which ensures that every network has unique set of contract
-identifiers.
 
 ### Installing WASM sources using `InvokeHostFunctionOp`
 
-WASM contract sources can be installed to the network without instantiating a
+WASM contract sources can be uploaded to the network without instantiating a
 contract via `InvokeHostFunctionOp`(defined in [CAP-0046-04](./cap-0046-04.md))
-with `HOST_FUNCTION_TYPE_INSTALL_CONTRACT_CODE` host function type.
+with `HOST_FUNCTION_TYPE_UPLOAD_CONTRACT_WASM` host function type in
+`hostFunction`.
 
-This function accepts `InstallContractCodeArgs` struct that contains the WASM
+This function accepts `opaque wasm<>` that contains the WASM
 contract code.
 
-Installed contracts are stored in `ContractCodeEntry` ledger entries. These
-entries are keyed by the hash of `InstallContractCodeArgs` used to install
+Uploaded contracts are stored in `ContractCodeEntry` ledger entries. These
+entries are keyed by the hash of the WASM used to upload
```

Comment:
nit: Could you please replace 'WASM' with 'Wasm' everywhere in the doc? WASM is not an official spelling, see https://webassembly.org/

Reply:
Done